scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Bash script to test the ast-import option of the compiler by
+# first exporting a .sol file to JSON, then loading it into the compiler
+# and exporting it again. The second JSON should be identical to the first
+READLINK=readlink
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    READLINK=greadlink
+fi
+IMPORT_TEST_TYPE=${1}
+REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
+SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+
+SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
+SEMANTICTESTS_DIR="${REPO_ROOT}/test/libsolidity/semanticTests"
+ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
+
+# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
+# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
+
+FAILED=0
+UNCOMPILABLE=0
+TESTED=0
+
+if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
+    echo "Test directory not empty. Skipping!"
+    exit 1
+fi
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local nth_input_file="$1"
+    IFS=" " read -r -a all_input_files <<< "$2"
+
+    if $SOLC --bin "$nth_input_file" "${all_input_files[@]}" > /dev/null 2>&1
+    then
+        ! [[ -e stderr.txt ]] || { echo "stderr.txt already exists. Refusing to overwrite."; exit 1; }
+
+        if [ "${IMPORT_TEST_TYPE}" == "ast" ]
+        then
+            # save exported json as expected result (silently)
+            $SOLC --combined-json ast --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> /dev/null
+            # import it, and export it again as obtained result (silently)
+            if ! $SOLC --import-ast --combined-json ast --pretty-json expected.json > obtained.json 2> stderr.txt
+            then
+                # For investigating, use exit 1 here so the script stops at the
+                # first failing test
+                # exit 1
+                FAILED=$((FAILED + 1))
+                echo -e "ERROR: AST reimport failed for input file $nth_input_file"
+                echo
+                echo "Compiler stderr:"
+                cat ./stderr.txt
+                echo
+                echo "Compiler stdout:"
+                cat ./obtained.json
+                return 1
+            fi
+            set +e
+            DIFF="$(diff expected.json obtained.json)"
+            set +e
```

Comment:
```suggestion
            DIFF="$(diff expected.json obtained.json || true)"
```
There are also a few more places like this below.

Reply:
Script uses now newly added `diff_files` from `common.sh`.